File: update_versions.py
# -*- coding: utf-8 -*-
import subprocess, json, sys, io

sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8', errors='replace')

SERVER = 'root@120.55.5.220'
REMOTE_DL = '/root/cloud-server/public/downloads/'

# Read current versions.json
r = subprocess.run(
    ['ssh', SERVER, f'cat {REMOTE_DL}versions.json'],
    capture_output=True, text=True, encoding='utf-8', errors='replace'
)

current = {}
if r.returncode == 0 and r.stdout.strip():
    try:
        current = json.loads(r.stdout)
        print('Current versions.json:', json.dumps(current, indent=2, ensure_ascii=False))
    except:
        print('Failed to parse existing versions.json')

# Create new version entry
new_version = {
    "version": "1.73.73",
    "releaseDate": "2026-06-14T22:00:00.000Z",
    "description": "修复: loadPrompts模块缺失问题",
    "files": {
        "win": {
            "installer": {
                "name": "灵境 Setup 1.73.73.exe",
                "size": 194595197,
                "sha512": "",
                "blockMap": "灵境 Setup 1.73.73.exe.blockmap"
            },
            "portable": {
                "name": "LingJing-Portable-1.73.73-win-x64.exe",
                "size": 194253710,
                "sha512": ""
            }
        }
    }
}

# Update versions array
if 'versions' not in current:
    current['versions'] = []

# Remove old 1.73.73 entry if exists
current['versions'] = [v for v in current['versions'] if v.get('version') != '1.73.73']
current['versions'].insert(0, new_version)

# Optionally set latest
current['latest'] = "1.73.73"

updated_json = json.dumps(current, indent=2, ensure_ascii=False)

# Write to temp file, then scp to server
tmp_local = r'D:\lingjing-ide\versions_v17373.json'
with open(tmp_local, 'w', encoding='utf-8') as f:
    f.write(updated_json)

print('\nNew versions.json:', updated_json)

# Upload
r = subprocess.run(
    ['scp', tmp_local, f'{SERVER}:{REMOTE_DL}versions.json'],
    capture_output=True, text=True, encoding='utf-8', errors='replace'
)
if r.returncode == 0:
    print('\n✅ versions.json uploaded!')
else:
    print(f'\n❌ Upload failed: {r.stderr[:200]}')

File: tools/update-versions.js
const fs = require('fs');
const data = JSON.parse(fs.readFileSync('/var/www/lingjing/versions.json', 'utf8'));

// Update latest
data.latest = '1.71.1';

// Add v1.71.1 entry
const newEntry = {
  version: '1.71.1',
  status: 'published',
  releaseDate: '2026-06-07',
  releaseNotes: '4项遗留审计缺陷修复(BUG-007/008/014/DB-002) + 版本号统一升级',
  platforms: {
    'win-x64': {
      sha512: '9e6922d0c12cedf4b8163c69785b5eb7109a4459a530f192b9f92b3a3e1f1354b09d57164d1c1a0641fc9a28842f69c582368b9130e3b13fbe3224caa870aab9',
      size: 142398152
    },
    'win-x64-portable': {
      sha512: 'e5415fb215fa245ffe2898383629e932042a0627fb6bd055a2355e97456ff2d6e81f7cc2d6d9342f8269bd30fc0eebc433a7b4048973d158462e2d8b2a4dfb3a',
      size: 142170461
    }
  },
  files: {
    'win-x64': 'https://ide.zhejiangjinmo.com/downloads/灵境 Setup 1.71.1.exe',
    'win-x64-portable': 'https://ide.zhejiangjinmo.com/downloads/LingJing-Portable-1.71.1-win-x64.exe'
  }
};

// Insert at top
data.versions.unshift(newEntry);

// Write back
fs.writeFileSync('/var/www/lingjing/versions.json', JSON.stringify(data, null, 2));
console.log('OK: latest=' + data.latest + ' count=' + data.versions.length);
